resource/hlapi/bindings/c++/reapi_cli_impl.hpp

```diff
@@ -97,7 +99,29 @@ int reapi_cli_t::match_allocate (void *h, bool orelse_reserve,
 
         R = o.str ();
 
-        if ( (rc = gettimeofday (&end_time, NULL)) < 0) {
+
+        reserved = (at != 0)? true : false; 
+        st = (reserved)? 
+                    job_lifecycle_t::RESERVED : job_lifecycle_t::ALLOCATED; 
+        if (reserved) 
+            rq->set_reservation (jobid); 
+        else
+            rq->set_allocation (jobid);
+ 
+        job_info = std::make_shared<job_info_t> (jobid, st, at, "", "", ov); 
+        if (job_info == NULL) {
```

Comment:
The original location wrapped this in a `try...catch`. The `make_shared` [documentation](https://en.cppreference.com/w/cpp/memory/shared_ptr/make_shared) mentions that it: "may throw [std::bad_alloc] or any exception thrown by the constructor of T. If an exception is thrown, the functions have no effect." I like the check for `NULL` since that appears to handle more potential failure modes. However, I think it's better to check for `nullptr`:

```suggestion
        if (job_info == nullptr) {
````